flutter_flexible_sync/lib/main.dart

```diff
@@ -23,8 +24,17 @@ Future<Realm> createRealm(String appId, CollectionType collectionType) async {
   final appConfig = AppConfiguration(appId);
   final app = App(appConfig);
   final user = await app.logIn(Credentials.anonymous());
-
-  final flxConfig = Configuration.flexibleSync(user, [Task.schema], path: await absolutePath("db_${collectionType.name}.realm"));
+  final Completer<void> completer = Completer<void>();
+  final flxConfig = Configuration.flexibleSync(
+    user,
+    [Task.schema],
+    path: await absolutePath("db_${collectionType.name}.realm"),
+    syncErrorHandler: (SyncError e) {
+      if (e.message!.startsWith("Host not found") && !completer.isCompleted) {
+        completer.complete();
+      }
```

Comment:
I think that's pretty unreliable - there are a bunch of http errors that may indicate that a connection is impossible and those may differ between different OS-es. If we want to show best practices, we should probably make this example more comprehensive and pull in the connectivity plugin as well as display some banner indicating the status of sync.

Reply:
Yes, it is unreliable. It is not a good way to check the connection in `syncErrorHandler`. Actually we don't need to check the connectivity anymore. `waitForSynchronization` doesn't lock the process with `realm 0.7.0+rc` in case of no connection. I' closing this PR.